### PR TITLE
Confluence APIv2 

### LINKED ISSUE #1205: Upgrade to Confluence REST API v2
In order to continue the support for Confluence Cloud we must migrate our integration to make use of REST API v2 until 01.01.2024 **latest** (see [reference](https://community.developer.atlassian.com/t/deprecating-many-confluence-v1-apis-that-have-v2-equivalents/66883)

### PR DESCRIPTION
**Pull request by PacoVK**
# Upgrade to Confluence APIv2

## What?
We need to upgrade to the Confluence API v2 until 01.24 latest with a lot of breaking changes. See  #1205 
We must be backwards compatible to ensure older Confluence versions can still be a target for tasks like `publishToConfluence`

In #1208  there is already a sample on how we could finally separate tools and tasks. Adhering this gives a bit more convenience and flexibility to implement the solution. Hence, this PR already is follows what #1208 suggests, but this is not a must and the implementation can be moved towards the current structure.

## The idea
Basically the idea is to introduce Confluence REST Clients that hide the actual API. The user can switch according her needs between the version that is required. 

### All Submissions:

* [x] Did you update the `changelog.adoc`?
* [x] Does your PR affect the documentation?
* [x] If yes, did you update the documentation or create an issue for updating it?

### CODE REVIEW

**Comment by PacoVK** [comment]
The first question to be answered is, should we stick to the current project structure or not

**Comment by rdmueller** [comment]
> The first question to be answered is, should we stick to the current project structure or not

What do you mean by "project structure"?

Shall we formulate this as ADR?

**Comment by rdmueller** [comment]
Please advice me how I can help best

**Comment by PacoVK** [comment]
Yeah, well I think an ADR was a good hint so we and others can have a discussion. I try to manage it to come up with something 👍

File: build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ buildscript {
                 details.because 'jcenter redirect'
                 System.out.println ">>>>>>>>> fix grolifant"
             }
+            //TODO is this really needed here?
```
Comment (rdmueller) [inline build.gradle:12]: this is a safety net which helped with when jcenter was gone. Maybe we can remove it by now, but it also does no harm

Comment (PacoVK) [reply build.gradle:12]: Yeah I remember and it makes sense I just wondered if it makes sense to go with 0.7.1 instead of force downgrade with this script. Because currently the versions file says 0.7.2

Comment (rdmueller) [reply build.gradle:12]: yeah, I am also nut sure if we are not already over this...

Comment (PacoVK) [reply build.gradle:12]: i'll keep that comment, but i will exceed the scope here

**Comment by PacoVK** [comment]
Now pending: the docs need some updates

**Comment by PacoVK** [comment]
@rdmueller FYI now, that i consider it ready, i will create the 3.1.0-rc1 in the next days 🏎️

**Comment by rdmueller** [comment]
These are great news! We also made progress with the draw.io png support. I hope I can add it vefore you create the 3.1.0 release 😁
Shall we create the release together?

**Comment by PacoVK** [comment]
sure, no pressure. Just let me know, if you are done. For the time beeing i suggest to freeze the confluence script for now. 
Or did you mean the 3.1.0 release or do you want to include it already in the 3.1.0-RC1?

**Comment by PacoVK** [comment]
closes #1265

**Comment by PacoVK** [comment]
closes #1254

**Review by rdmueller — APPROVED** [review]
LGTM
